### PR TITLE
Bug fix/skip around gather executors (#11945)

### PR DESCRIPTION
**Pull request by mchacki**
Backport of https://github.com/arangodb/arangodb/pull/11945

Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10650/

### CODE REVIEW

File: arangod/Aql/ExecutionBlockImpl.cpp [hunk 1]
```diff
@@ -1311,6 +1318,9 @@ ExecutionBlockImpl<Executor>::executeWithoutTrace(AqlCallStack stack) {
 
   LOG_QUERY("007ac", DEBUG) << "starting statemachine of executor " << printBlockInfo();
   while (_execState != ExecState::DONE) {
+    // We can never keep state in the skipCounter
+    TRI_ASSERT(clientCall.getSkipCount() == 0);
+    auto oldState = _execState;
```
Comment (jsteemann) [inline arangod/Aql/ExecutionBlockImpl.cpp:1323]: Please remove this line, as `oldState` is an unneeded variable and will cause a compile warning (or error, if warnings are propagated to errors).
```suggestion
```

Comment (mchacki) [reply arangod/Aql/ExecutionBlockImpl.cpp]: Thanks for spotting, left-over of debugging ;(

**Comment by mchacki** [comment]
Jenkins with unused variable removed:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10657/

**Comment by jsteemann** [comment]
Tests blue

**Pull request merged**